tools/SourceKit/lib/SwiftLang/SwiftSourceDocInfo.cpp

```diff
@@ -641,18 +641,21 @@ static bool passCursorInfoForModule(ModuleEntity Mod,
                        std::function<void(const RequestResult<CursorInfoData> &)> Receiver) {
   std::string Name = Mod.getName().str();
   std::string FullName = Mod.getFullName();
+
   CursorInfoData Info;
-  Info.Kind = SwiftLangSupport::getUIDForModuleRef();
-  Info.Name = Name;
-  Info.ModuleName = FullName;
-  if (auto IFaceGenRef = IFaceGenContexts.find(Info.ModuleName, Invok))
-    Info.ModuleInterfaceName = IFaceGenRef->getDocumentName();
-  Info.IsSystem = Mod.isSystemModule();
-  std::vector<StringRef> Groups;
+  CursorSymbolInfo Symbol;
+  Symbol.Kind = SwiftLangSupport::getUIDForModuleRef();
+  Symbol.Name = Name;
+  Symbol.ModuleName = FullName;
+  if (auto IFaceGenRef = IFaceGenContexts.find(Symbol.ModuleName, Invok))
+    Symbol.ModuleInterfaceName = IFaceGenRef->getDocumentName();
+  Symbol.IsSystem = Mod.isSystemModule();
   if (auto MD = Mod.getAsSwiftModule()) {
-    Info.ModuleGroupArray = ide::collectModuleGroups(const_cast<ModuleDecl*>(MD),
-                                                     Groups);
+    ide::collectModuleGroups(const_cast<ModuleDecl *>(MD),
+                             Symbol.ModuleGroupArray);
   }
+  Info.Symbols.push_back(Symbol);
```

Comment:
`push_back(std::move(Symbol))` - this type is huge and contains SmallVectors, so might actually matter.

Reply:
Oh nice catch! Thanks.

Reply:
If `Symbol` is huge, I'd do:
```cpp
CursorSymbolInfo &Symbol = Info.Symbols.emplace_back();
Symbol.Kind = ...;
Symbol.Name = ...;
```

Reply:
Hmm that's true, I could do that in the other two places as well. Thanks Rintaro!